pygmt/clib/conversion.py

```diff
@@ -162,11 +162,43 @@ def vectors_to_arrays(vectors):
     True
     >>> all(isinstance(i, np.ndarray) for i in arrays)
     True
+
     >>> data = [[1, 2], (3, 4), range(5, 7)]
     >>> all(isinstance(i, np.ndarray) for i in vectors_to_arrays(data))
     True
+
+    >>> import datetime
+    >>> import pytest
+    >>> pa = pytest.importorskip("pyarrow")
+    >>> vectors = [
+    ...     pd.Series(
+    ...         data=[datetime.date(2020, 1, 1), datetime.date(2021, 12, 31)],
+    ...         dtype="date32[day][pyarrow]",
+    ...     ),
+    ...     pd.Series(
+    ...         data=[datetime.date(2022, 1, 1), datetime.date(2023, 12, 31)],
+    ...         dtype="date64[ms][pyarrow]",
+    ...     ),
+    ... ]
+    >>> arrays = vectors_to_arrays(vectors)
+    >>> all(a.flags.c_contiguous for a in arrays)
+    True
+    >>> all(isinstance(a, np.ndarray) for a in arrays)
+    True
+    >>> all(isinstance(a.dtype, np.dtypes.DateTime64DType) for a in arrays)
+    True
     """
-    arrays = [as_c_contiguous(np.asarray(i)) for i in vectors]
+    arrays = []
```

Comment:
I haven't tried it, but maybe the following code is simpler:
```
dtypes = {
    "date32[day][pyarrow]": np.datetime64,
    "date64[ms][pyarrow]": np.datetime64,
}
arrays = []
for vector in vectors:
    vec_dtype = str(getattr(vector, "dtype", ""))
    array = np.asarray(dtypes.get(vec_dtype, None))
    arrays.append(as_c_contiguous(array))
```

Reply:
Oh nice, that's much simpler and makes it easier to extend to other dtypes in the future! Done at commit de08508d82fe2e5e79255542ca65193010320699.